### PR TITLE
ref(storages): Split events_ro into its own storage

### PR DESCRIPTION
**Pull request by lynnagara**
The `sentry_dist` vs `sentry_dist_ro` table selection is now
the responsibility of the dataset storage selector and does not
rely on a query processor to switch out the table name.
This is now implemented for the Discover dataset. Dataset storage
selection will also need to be rolled out to the Events dataset before
the ReadOnlyTableSelector can be removed.

Currently (in this PR) our new `event_ro` storage  is still part of the
events storage set. In a follow up, it will be moved to its own storage
set so that it can have a distinct cluster configuration. In prod, the
`sentry_dist_ro` table points at a different ClickHouse cluster to
`sentry_dist` so this is necessary to accurately reflect our current
configuration in settings.

### CODE REVIEW

File: snuba/datasets/discover.py [hunk 1]
```diff
@@ -55,7 +55,7 @@
 logger = logging.getLogger(__name__)
 
 
-def detect_table(
+def detect_dataset(
```
Comment (fpacifici) [inline snuba/datasets/discover.py:58]: Why was this renamed to detect_dataset ? We are dealing with one dataset here and multiple storages, the name would be misleading.

File: snuba/datasets/discover.py [hunk 2]
```diff
@@ -197,19 +199,27 @@ def __init__(
...
             )
-        )
+        else:
+            use_readonly_storage = (
+                state.get_config("enable_events_readonly_table", False)
+                and not request_settings.get_consistent()
+            )
+            return (
+                StorageAndMappers(self.__events_ro_table, self.__event_translator)
+                if use_readonly_storage
```
Comment (fpacifici) [inline snuba/datasets/discover.py:220]: Won't this potentially select an empty table on development environments and on prem? None is writing on events_ro there. Or do you plan to address that subsequently making the events_ro table only be created in distributed environments ?
In such case will we even have a local table definition for this storage ?

**Review by fpacifici — COMMENTED** [review]
Just one clarification on the plan otherwise, please rename `detect_dataset`.

File: snuba/datasets/discover.py (hunk 2 above)
Comment (lynnagara) [reply snuba/datasets/discover.py:220]: I don't think so. The local table name is the same in both storages: "sentry_local". The new storage picks the relevant name "sentry_local" or "sentry_dist_ro". It works the same way the old one picked between "sentry_local" and "sentry_dist".

Comment (fpacifici) [reply snuba/datasets/discover.py:220]: right.

File: snuba/datasets/discover.py (hunk 1 above)
Comment (lynnagara) [reply snuba/datasets/discover.py]: Reverted this

**Review by fpacifici — APPROVED** [review]
